stdlib/array.mli

```diff
@@ -45,8 +45,8 @@ external make : int -> 'a -> 'a array = "caml_make_vect"
    will modify all other entries at the same time.
 
    Raise [Invalid_argument] if [n < 0] or [n > Sys.max_array_length].
-   If the value of [x] is a floating-point number, then the maximum
-   size is only [Sys.max_array_length / 2].*)
+   On 32-bit machines, if the value of [x] is a floating-point number,
```

Comment:
Why the restriction to 32-bit machines? If I read `byterun/array.c:caml_make_vect` correctly, the size restriction applies no matter the bit-size. It's just that 64-bit machines have a very large Sys.max_array_length value.

Reply:
On the other hand, `-no-flat-float-array` systems are not concerned by the size halving. (I don't think this should be mentioned in the documentation as this mode is experimental.)

Reply:
The array size is adjusted by `Double_wosize` in the flat float array case (`byterun/array.c:caml_make_vect` line 315) : `Double_wosize` is `sizeof (double) / sizeof (value)`, so 1 on 64-bit arch, 2 on 32-bit.

Reply:
also *c.f*. the comment of `Sys.max_array_length`
```
val max_array_length : int
(** Maximum length of a normal array.  The maximum length of a float
    array is [max_array_length/2] on 32-bit machines and
    [max_array_length] on 64-bit machines. *)
```